src/nvim/path.c

```diff
@@ -2116,15 +2121,15 @@ int append_path(char *path, const char *to_append, size_t max_len)
   }
 
   // Glue both paths with a slash.
-  if (current_length > 0 && path[current_length-1] != '/') {
+  if (current_length > 0 && path[current_length-1] != PATHSEP) {
     current_length += 1;  // Count the trailing slash.
 
     // +1 for the NUL at the end.
     if (current_length + 1 > max_len) {
       return FAIL;
     }
 
-    STRCAT(path, "/");
+    STRCAT(path, PATHSEPSTR);
```

Comment:
`path_expand`/`do_path_expand` is our replacement of Vim's `mch_expandpath`/`dos_expandpath`/etc.   A brief inspection does not reveal any direct handling of `p_ssl` (`'shellslash'` option) in Vim's `dos_expandpath`. 

I would think we should handle `p_ssl` here, but maybe it gets fixed up before reaching the user?